nco/custom.py

```diff
@@ -0,0 +1,400 @@
+})
+
+# numpy type names
+NP_TYPES = dict({
+    'float32': 'f',
+    'float64': 'd',
+    'int32': 'i',
+    'int16': 's',
+    'char': 'c',
+    'byte': 'b',
+    'ubyte': 'ub',
+    'int8': 'b',
+    'uint8': 'ub',
+    'uint16': 'us',
+    'uint32': 'ui',
+    'int64': 'll',
+    'uint64': 'ull',
+    'string': 'sng'
+})
+
+NP_TYPESNP = dict({
+    'float32': np.float32,
+    'float': np.float32,
+    'f': np.float32,
+    'float64': np.float64,
+    'double': np.float64,
+    'd': np.float64,
+    'int32': np.int32,
+    'i': np.int32,
+    'l': np.int32,
+    'int16': np.int16,
+    's': np.int16,
+    'str': str,
+    'char': str,
+    'c': str,
+    'string': str,
+    'sng': str,
+    'byte': np.byte,
+    'b': np.byte,
+    'ubyte': np.ubyte,
+    'ub': np.ubyte,
+    'int8': np.byte,
+    'uint8': np.ubyte,
+    'uint16': np.uint16,
+    'us': np.uint16,
+    'uint32': np.uint32,
+    'u': np.uint32,
+    'ui': np.uint32,
+    'ul': np.uint32,
+    'int64': np.int64,
+    'll': np.int64,
+    'uint64': np.uint64,
+    'ull': np.uint64
+})
+
+__prog__ = os.path.splitext(os.path.basename(__file__))[0]
+
+
+# wrapper to -a ncatted command line switch
+# command line swtich looks like
+#
+# -a,att_nm,var_nm,mode,att_typ,att_val
+# mode = a,c,d,m,o,n (append, create, delete, modify, overwrite, nappend)
+# att_typ = f,d,l/i,s,c,b,ub,us,u,ll,ull,sng
+class atted(object):
+    def __init__(self, mode='overwrite', att_name="", var_name="", value=None, stype=None, **kwargs):
+        mode = kwargs.pop('mode', mode)
+        att_name = kwargs.pop('att_name', att_name)
+        var_name = kwargs.pop('var_name', var_name)
+        value = kwargs.pop('value', value)
+        stype = kwargs.pop('stype', stype)
+
+        if mode in VALID_MODES:
+            mode = VALID_MODES[mode]
+        elif mode in VALID_MODES.values():
+            pass
+        else:
+            raise Exception('mode "{0}" not found'.format(mode))
+
+        if not att_name:
+            raise ValueError('att_name is required')
+
+        # set member defaults
+        self.mode = mode
+        self.att_name = att_name
+        self.var_name = var_name
+        self.np_type = None
+        self.np_value = None
+
+        # dont bother about type & value
+        if self.mode == "d":
+            return
+
+        # deal with string quirk 
+        # if user specifies type 'string' or 'sng' with a single character string 
+        # the the output type should be 'sng and NOT 'c'. To do this we put the single string
+        # inside a list the the prnOptions() method treats it as it would a list of strings
+        if isinstance(value, str) and stype in ['string', 'sng']:
+            value = [value]
+
+        # see if value is iterable   
+        try:
+            if isinstance(value, str):
+                raise TypeError("str is not iterable (for our purposes")
+            it = iter(value)
+            inputType = type(next(it))
+            bIterable = True
```

Comment:
I'll just say this once but you need to correct it everywhere. Variable names should use `lower_case_with_underscores` syntax, class names should use `CamelCase` syntax.